lib/nrf_modem_lib/trace_backends/rtt/rtt.c

```diff
@@ -18,7 +19,13 @@ static trace_backend_processed_cb trace_processed_callback;
 
 int trace_backend_init(trace_backend_processed_cb trace_processed_cb)
 {
-	const int segger_rtt_mode = SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL;
+	int segger_rtt_mode;
+
+#if CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT_SKIP_IF_FIFO_FULL
+	segger_rtt_mode = SEGGER_RTT_MODE_NO_BLOCK_SKIP;
+#else
+	segger_rtt_mode = SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL;
+#endif /* CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT_SKIP_IF_FIFO_FULL */
```

Comment:
We should make sure that the error is handled correctly by the trace thread too, by exiting.

Reply:
The assumptions above are not entirely correct. With segger RTT mode set to `SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL`, the call to `SEGGER_RTT_WriteNoLock` itself does not return if the FIFO is full. Hence, the application core does not go to sleep. If setting the RTT mode to `SEGGER_RTT_MODE_NO_BLOCK_SKIP` it returns with 0, indicating that no data is written. The issue here, though, is that we have no callback for when there is space in the FIFO or so that can allow us to wait for a semaphore etc. So we either have to drop the data or keep polling the RTT buffer/trying to write new data. 

Though, I am now thinking, that if the application sets the modem trace level to off using `nrf_modem_lib_trace_level_set(0)`, we could use this together with the error from the RTT backend to wait until modem traces are enabled again. Then it is likely that an RTT client is connected and we can proceed with the tracing. However, this would only work if the application is using `nrf_modem_lib_trace_level_set()` and not the AT command directly. 

> We should make sure that the error is handled correctly by the trace thread too, by exiting.

We cannot exit the trace thread, as the user would need to be able to connect RTT client at a later time, then resume tracing from that point onwards.